packages/bolt-connection/test/bolt/bolt-protocol-v5x8.test.js

```diff
@@ -0,0 +1,1579 @@
+    ])
+    const txConfig = new TxConfig({
+      timeout: 5000,
+      metadata: { x: 1, y: 'something' }
+    })
+    const recorder = new utils.MessageRecordingConnection()
+    const protocol = new BoltProtocolV5x8(recorder, null, false)
+    utils.spyProtocolWrite(protocol)
+
+    const observer = protocol.beginTransaction({
+      bookmarks,
+      txConfig,
+      database,
+      mode: WRITE,
+      impersonatedUser
+    })
+
+    protocol.verifyMessageCount(1)
+    expect(protocol.messages[0]).toBeMessage(
+      RequestMessage.begin({ bookmarks, txConfig, database, mode: WRITE, impersonatedUser })
+    )
+    expect(protocol.observers).toEqual([observer])
+    expect(protocol.flushes).toEqual([true])
+  })
+
+  it('should return correct bolt version number', () => {
+    const protocol = new BoltProtocolV5x8(null, null, false)
+
+    expect(protocol.version).toBe(5.8)
+  })
+
+  it('should update metadata', () => {
+    const metadata = { t_first: 1, t_last: 2, db_hits: 3, some_other_key: 4 }
+    const protocol = new BoltProtocolV5x8(null, null, false)
+
+    const transformedMetadata = protocol.transformMetadata(metadata)
+
+    expect(transformedMetadata).toEqual({
+      result_available_after: 1,
+      result_consumed_after: 2,
+      db_hits: 3,
+      some_other_key: 4
+    })
+  })
+
+  it('should initialize connection', () => {
+    const recorder = new utils.MessageRecordingConnection()
+    const protocol = new BoltProtocolV5x8(recorder, null, false)
+    utils.spyProtocolWrite(protocol)
+
+    const clientName = 'js-driver/1.2.3'
+    const boltAgent = {
+      product: 'neo4j-javascript/5.8',
+      platform: 'netbsd 1.1.1; Some arch',
+      languageDetails: 'Node/16.0.1 (v8 1.7.0)'
+    }
+    const authToken = { username: 'neo4j', password: 'secret' }
+
+    const observer = protocol.initialize({ userAgent: clientName, boltAgent, authToken })
+
+    protocol.verifyMessageCount(2)
+    expect(protocol.messages[0]).toBeMessage(
+      RequestMessage.hello5x3(clientName, boltAgent)
+    )
+    expect(protocol.messages[1]).toBeMessage(
+      RequestMessage.logon(authToken)
+    )
+
+    expect(protocol.observers.length).toBe(2)
+
+    // hello observer
+    const helloObserver = protocol.observers[0]
+    expect(helloObserver).toBeInstanceOf(LoginObserver)
+    expect(helloObserver).not.toBe(observer)
+
+    // login observer
+    const loginObserver = protocol.observers[1]
+    expect(loginObserver).toBeInstanceOf(LoginObserver)
+    expect(loginObserver).toBe(observer)
+
+    expect(protocol.flushes).toEqual([false, true])
+  })
+
+  it.each([
+    'javascript-driver/5.8.0',
+    '',
+    undefined,
+    null
+  ])('should always use the user agent set by the user', (userAgent) => {
+    const recorder = new utils.MessageRecordingConnection()
+    const protocol = new BoltProtocolV5x8(recorder, null, false)
+    utils.spyProtocolWrite(protocol)
+
+    const boltAgent = {
+      product: 'neo4j-javascript/5.8',
```

Comment:
You're entirely correct. These lines were left un-bumped since their introduction in 5.6 driver testing bolt 5.3, then when creating the bolt 5.7 test I ended up bumping them. I'll correct this for other bolt tests in a separate PR to not clutter this one.